base_technical_user/models/models.py

```diff
@@ -12,9 +12,21 @@ class Base(models.AbstractModel):
 
     def sudo_tech(self, raise_if_missing=False):
         self_sudoer = self
-        tech_user = self.env.user.company_id.user_tech_id
+        tech_user = self.env.company.user_tech_id
         if tech_user:
             self_sudoer = self.with_user(tech_user.id)
+            # We restrict the allowed companies to the one of the tech user
+            allowed_company_ids = self.env.context.get("allowed_company_ids")
+            # TODO: Is any(...) part necessary as we can consider company should always be
```

Comment:
if the check is not necessary then even the check on `allowed_company_ids` seems useless because it comes from the the same context of `self` :thinking:

Reply:
In allowed_company_ids, you can have several companies depending on e.g. the user interface behaviour.

The problem is this check: https://github.com/odoo/odoo/blob/14.0/odoo/api.py#L585

As user that launches action can have access to company A and B. But, tech user has only access to company A.
So, allowed_company_ids comes with [A, B] and raises.

Maybe, we can limit the checks indeed and only force the value of allowed_company_ids. What do you think ?

Reply:
I see, thanks for the detail. Yes, I think it makes sense to simply clean them up and keep only the ones available.